Makefile

```diff
@@ -18,6 +18,9 @@ build: build-web build-go
 build-go:
 	go build ./...
 
+bin/viam$(BINPREFIX)-$(GOOS)-$(GOARCH): ./cli/viam
+	go build -ldflags="-X 'go.viam.com/rdk/config.Version=$(TAG_VERSION)' -s -w" -tags osusergo,netgo -o $@ ./$<
+
```

Comment:
So very clever what you did here! Especially using the deps as an argument. But it's not easily useable by humans, and still leaves a lot in the workflow (instead of the makefiles.) I harp on this because it's EASY to change a makefile and test that it still works, even just in a PR. But testing a workflow change is more complicated, so it's best (in my opinion) to keep as much logic as possible in the Makefile.

Forgive me for making some significant code edit suggestions here, but... I think something like this (combined with the other suggestions I made to use it) simplifies the use of things (and is a big clearer to read.)

Please feel free to push back on this if you think it's a bad idea or if I'm stepping on your toes too much.

```suggestion
GOOS ?= $(shell go env GOOS)
GOARCH ?= $(shell go env GOARCH)
bin/$(GOOS)-$(GOARCH)/viam-cli:
	go build $(LDFLAGS) -tags osusergo,netgo -o $@ ./cli/viam

.PHONY: cli
cli: bin/$(GOOS)-$(GOARCH)/viam-cli

.PHONY: cli-ci
cli-ci: bin/$(GOOS)-$(GOARCH)/viam-cli
	if [[ -n "$(CI_RELEASE)" ]]; then \
		mkdir -p bin/deploy-ci/; \
		cp $< bin/deploy-ci/viam-cli-$(CI_RELEASE)-$(GOOS)-$(GOARCH); \
	fi

```
Now I can just type `make cli` and get the local binary I expect. But setting the GOOS/GOARCH variables let's it build cross platform as well. Lastly, by setting CI_RELEASE and using `make cli-ci` it does the renaming for deployment too.

Also, the variable setting (for GOOS and GOARCH) could be at the top of the file. Eventually I'd like to update it so that BIN_OUTPUT_PATH itself uses the GOOS-GOARCH format, rather than the shell stuff it uses now, but this is in prep for doing that later I think.